docs/DeveloperGuide.md

```diff
@@ -1,9 +1,263 @@
-# Developer Guide
+# TraceYourProj's Developer Guide
 
 ## Design & implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Design
+{insert design}
+
+### Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add projects and resources
+**Proposed Implementation**
+
+Given below is an example usage scenario and how the add mechanism behaves at each step.
+
+**Step 1**:
+The user launches the application. The project list of the application will be initialized as an empty list.
+
+**Step 2**:
+The user executes the following commands in order to add a resources to a project.
+
+>add p/CS2113 url/www.traceyourproj.com d/Project Website
+
+The following sequence diagram shows how the add operation works:
+![add_png](puml_img/Add.png)
+
+> Note: P2:Project is defined as an entity that is already existing in the project list ArrayList<Project>, the position
+> of which should be lower than ArrayList<Project>. However, due to a limitation in PlantUML, the position of P3 is 
+> higher than ArrayList<Project>.
+
+**Design Consideration**
+
+Aspect: How add executes
+* Alternative 1 (current choice):
+  * The add feature will do a loop through all the projects in the ArrayList projects to see if the project has already
+    existed. For each project, it will loop through all the resources to see if a resource with the same URL has already
+    existed.
+    
+    > If the project does not exist in the ArrayList:
+    >   * Create a new project and append the resource into its resource list.
+    
+    > If the project does exist:
+    > * If resource with the same URL exists:
+    >   * Prompt the user that the resource with the same URL has already existed
+        in the project's resource list.
+    > * If resource with the same URL does not exist:
+    >   * Append the new resource into the project resource list.
+
+* Alternative 2 (none)
+---
+### List all projects & their respective resources feature
```

Comment:
Would it be better to standardize all your section headings? This has the word feature at the end but some of the other headings don't.